File: GestionRessourcesInfo/src/main/java/com/example/GestionRessourcesInfo/model/Rapport.java
package com.example.GestionRessourcesInfo.model;

import com.example.GestionRessourcesInfo.model.enums.TypeRapport;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Rapport {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Enumerated(EnumType.STRING)
    private TypeRapport type;
    private Date dateGeneration;
    @Lob
    private String contenu;

    @ManyToOne
    @JoinColumn(name = "administrateurIT_id")
    private AdministrateurIT administrateurIT;

    @ManyToOne
    @JoinColumn(name = "technicienIT_id")
    private TechnicienIT technicienIT;

}

File: FrontEnd/front-endSi/src/app/models/rapport.ts
import {TypeRapport} from "./enums/type-rapport";
import {TechnicienIT} from "./technicien-it";
import {AdministrateurIT} from "./administrateur-it";

export interface Rapport {
  id?: number;
  type: TypeRapport;
  dateGeneration: string;
  contenu: string;
  administrateurIT: { id: number };
  technicienIT?: { id: number };
}
